src/DDTrace/Integrations/OpenAI/OpenAIIntegration.php

```diff
@@ -0,0 +1,1185 @@
+
+            'openai.response.object' => $response['object'] ?? null,
+        ];
+
+        switch ($methodName) {
+            case 'createModeration':
+                $tags += self::createModerationResponseExtraction($response);
+                break;
+
+            case 'createCompletion':
+            case 'createChatCompletion':
+                $tags += self::commonCreateResponseExtraction($response);
+                break;
+
+            case 'listFiles':
+            case 'listFineTunes':
+            case 'listFineTuneEvents':
+                $tags += self::commonListCountResponseExtraction($response);
+                break;
+
+            case 'createEmbedding':
+                $tags += self::createEmbeddingResponseExtraction($response);
+                break;
+
+            case 'createFile':
+            case 'retrieveFile':
+                $tags += self::createRetrieveFileResponseExtraction($response);
+                break;
+
+            case 'deleteFile':
+                $tags += self::deleteFileResponseExtraction($response);
+                break;
+
+            case 'downloadFile':
+                $tags += self::downloadFileResponseExtraction($response);
+                break;
+
+            case 'createFineTune':
+            case 'retrieveFineTune':
+            case 'cancelFineTune':
+                $tags += self::commonFineTuneResponseExtraction($response);
+                break;
+
+            case 'createTranscription':
+            case 'createTranslation':
+                $tags += self::createAudioResponseExtraction($response);
+                break;
+
+            case 'createImage':
+            case 'createImageEdit':
+            case 'createImageVariation':
+                $tags += self::commonImageResponseExtraction($response);
+                break;
+
+            case 'listModels':
+                $tags += self::listModelsResponseExtraction($response);
+                break;
+
+            case 'retrieveModel':
+                $tags += self::retrieveModelResponseExtraction($response);
+                break;
+        }
+
+        self::applyTags($span, $tags);
+
+        self::sendMetrics(
+            span: $span,
+            headers: $headers,
+            duration: $span->getDuration(),
+            promptTokens: (int)($response['usage']['prompt_tokens'] ?? 0),
+            completionTokens: (int)($response['usage']['completion_tokens'] ?? 0)
+        );
+
+        self::sendLog(
+            logger: $logger,
+            span: $span,
+            methodName: $methodName,
+            error: $span->exception ? true : false
+        );
+
+        self::applyPromptAndCompletionSampling($span);
+    }
+
+    public static function getLogTags(
+        SpanData $span,
+        string   $methodName
+    ): array
+    {
+        $tags = [
+            'env' => \dd_trace_env_config('DD_ENV'),
+            'version' => \dd_trace_env_config('DD_VERSION'),
```

Comment:
```suggestion
            'env' => $span->env,
            'version' => $span->version,
```
No point in fetching from config - it's inherited and that's the important source.